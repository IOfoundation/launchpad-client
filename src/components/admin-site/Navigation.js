import React from 'react';
import {Link} from 'react-router';
import {withStyles} from '@material-ui/core/styles';
import {PropTypes} from 'prop-types';

const Navigation = props => {
  const {classes} = props;

  return (
    <div className={classes.container}>
      <h3 className={classes.title}>{'Your organization'}</h3>
      <p className={classes.description}>
        {"Governor's Office of Business and Economic Development"}
      </p>
      <ul className={classes.navigation}>
        <li>
          <Link
            activeStyle={activeStyles}
            className={classes.link}
            to="/admin/profile"
          >
            {'Profile'}
          </Link>
        </li>
        <li>
          <Link
            activeStyle={activeStyles}
            className={classes.link}
            to="/admin/location"
          >
            {'Locations'}
          </Link>
        </li>
        <li>
          <Link
            activeStyle={activeStyles}
            className={classes.link}
            to="/admin/services"
          >
            {'Services'}
          </Link>
        </li>
        <li>
          <a className={classes.link}>{'Blog Posts'}</a>
        </li>
        <li>
          <a className={classes.link}>{'Events'}</a>
        </li>
      </ul>
      <h3 className={`${classes.title} m-top-16`}>{'Status'}</h3>
      <p className={classes.description}>{'Published'}</p>
      <button className="btn btn__red">{'Unpublish'}</button>
    </div>
  );
};

const activeStyles = {
  color: '#00ba81',
  opacity: 1,
  fontFamily: '"proxima-nova-bold", Georgia, sans-serif',
};

const styles = () => {
  return {
    container: {
      background: 'black',
      margin: '68px 0 24px',
      padding: '24px',
    },
    title: {
      fontFamily: '"proxima-nova-bold", Georgia, sans-serif',
      fontSize: '14px',
      lineHeight: '20px',
      marginBottom: '8px',
    },
    description: {
      fontFamily: '"proxima-nova-thin", Georgia, sans-serif',
      fontSize: '16px',
      lineHeight: '24px',
      marginBottom: '24px',
    },
    navigation: {
      listStyleType: 'none',
      margin: 0,
      padding: 0,
      borderBottom: '1px solid #1F1F1F',
      '& li': {
        marginBottom: '16px',
      },
    },
    link: {
      color: 'white',
      opacity: 0.75,
      '&:hover': {
        color: '#00BA81',
        opacity: 1,
      },
    },
  };
};

Navigation.propTypes = {
  classes: PropTypes.shape({
    container: PropTypes.string,
    description: PropTypes.string,
    link: PropTypes.string,
    navigation: PropTypes.string,
    title: PropTypes.string,
  }),
};

export default withStyles(styles)(Navigation);
