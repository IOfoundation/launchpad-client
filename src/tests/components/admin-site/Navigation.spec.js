import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

import Navigation from '../../../components/admin-site/Navigation';

const initialState = {
  adminProfile: {
    updatedOrganization: {
      data: [],
      loading: false,
      success: false,
      errors: [],
    },
    publishStatus: null,
    publishSuccess: false,
    publishLoading: false,
    publishErrors: [],
  },
  user: {
    organizationId: 1,
  },
  businesses: {
    organization: {},
  },
};
const mockStore = configureStore([thunk]);
let store;
let wrapper;
const props = {
  location: {
    pathname: '/admin/profile',
  },
};

describe('Admin-site <Navigation />', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <Navigation {...props} />
      </Provider>
    );
  });

  it('renders snapshot of Navigation', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
