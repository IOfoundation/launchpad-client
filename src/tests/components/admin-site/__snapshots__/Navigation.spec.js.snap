// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin-site <Navigation /> renders snapshot of Navigation 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(WithStyles(withRouter(Navigation)))
    location={
      Object {
        "pathname": "/admin/profile",
      }
    }
  >
    <WithStyles(withRouter(Navigation))
      auth={null}
      errors={Array []}
      location={
        Object {
          "pathname": "/admin/profile",
        }
      }
      organization={Object {}}
      organizationId={1}
      profile={
        Object {
          "updateCompany": [Function],
          "updatePublishStatus": [Function],
        }
      }
      schemaFail={false}
      schemaValidated={false}
      snackbar={
        Object {
          "hideSnackbar": [Function],
          "showSnackbar": [Function],
          "testingSnackbar": [Function],
        }
      }
      success={false}
      updatedOrganization={Object {}}
      yupActions={
        Object {
          "validateSchema": [Function],
        }
      }
    >
      <withRouter(Navigation)
        auth={null}
        classes={
          Object {
            "addBorder": "withRouter-Navigation--addBorder-5",
            "container": "withRouter-Navigation--container-1",
            "description": "withRouter-Navigation--description-3",
            "link": "withRouter-Navigation--link-6",
            "navigation": "withRouter-Navigation--navigation-4",
            "title": "withRouter-Navigation--title-2",
          }
        }
        errors={Array []}
        location={
          Object {
            "pathname": "/admin/profile",
          }
        }
        organization={Object {}}
        organizationId={1}
        profile={
          Object {
            "updateCompany": [Function],
            "updatePublishStatus": [Function],
          }
        }
        schemaFail={false}
        schemaValidated={false}
        snackbar={
          Object {
            "hideSnackbar": [Function],
            "showSnackbar": [Function],
            "testingSnackbar": [Function],
          }
        }
        success={false}
        updatedOrganization={Object {}}
        yupActions={
          Object {
            "validateSchema": [Function],
          }
        }
      >
        <Navigation
          auth={null}
          classes={
            Object {
              "addBorder": "withRouter-Navigation--addBorder-5",
              "container": "withRouter-Navigation--container-1",
              "description": "withRouter-Navigation--description-3",
              "link": "withRouter-Navigation--link-6",
              "navigation": "withRouter-Navigation--navigation-4",
              "title": "withRouter-Navigation--title-2",
            }
          }
          errors={Array []}
          location={
            Object {
              "pathname": "/admin/profile",
            }
          }
          organization={Object {}}
          organizationId={1}
          profile={
            Object {
              "updateCompany": [Function],
              "updatePublishStatus": [Function],
            }
          }
          schemaFail={false}
          schemaValidated={false}
          snackbar={
            Object {
              "hideSnackbar": [Function],
              "showSnackbar": [Function],
              "testingSnackbar": [Function],
            }
          }
          success={false}
          updatedOrganization={Object {}}
          yupActions={
            Object {
              "validateSchema": [Function],
            }
          }
        >
          <div
            className="withRouter-Navigation--container-1"
          >
            <WithStyles(Modal)
              activeClicked={[Function]}
              cancelClicked={[Function]}
              modalClosed={[Function]}
              open={false}
              schemaFail={false}
              schemaValidated={false}
              statusPicked={false}
            >
              <Modal
                activeClicked={[Function]}
                cancelClicked={[Function]}
                classes={
                  Object {
                    "alert": "Modal-alert-7",
                    "btn": "Modal-btn-9",
                    "modal": "Modal-modal-8",
                    "title": "Modal-title-10",
                  }
                }
                modalClosed={[Function]}
                open={false}
                schemaFail={false}
                schemaValidated={false}
                statusPicked={false}
              >
                <WithStyles(Modal)
                  onClose={[Function]}
                  open={false}
                >
                  <Modal
                    BackdropComponent={[Function]}
                    classes={
                      Object {
                        "hidden": "MuiModal-hidden-12",
                        "root": "MuiModal-root-11",
                      }
                    }
                    closeAfterTransition={false}
                    disableAutoFocus={false}
                    disableBackdropClick={false}
                    disableEnforceFocus={false}
                    disableEscapeKeyDown={false}
                    disablePortal={false}
                    disableRestoreFocus={false}
                    hideBackdrop={false}
                    keepMounted={false}
                    manager={
                      ModalManager {
                        "data": Array [],
                        "handleContainerOverflow": true,
                        "hideSiblingNodes": true,
                        "modals": Array [],
                      }
                    }
                    onClose={[Function]}
                    open={false}
                  />
                </WithStyles(Modal)>
              </Modal>
            </WithStyles(Modal)>
            <h3
              className="withRouter-Navigation--title-2"
            >
              Your organization
            </h3>
            <p
              className="withRouter-Navigation--description-3"
            >
              Your organization display name has not been set.
            </p>
            <ul
              className="withRouter-Navigation--navigation-4 withRouter-Navigation--addBorder-5"
            >
              <li>
                <Link
                  activeStyle={
                    Object {
                      "color": "#00ba81",
                      "fontFamily": "\\"proxima-nova-bold\\", Georgia, sans-serif",
                      "opacity": 1,
                    }
                  }
                  className="withRouter-Navigation--link-6"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/admin/profile"
                >
                  <a
                    className="withRouter-Navigation--link-6"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Profile
                  </a>
                </Link>
              </li>
              <li>
                <Link
                  activeStyle={
                    Object {
                      "color": "#00ba81",
                      "fontFamily": "\\"proxima-nova-bold\\", Georgia, sans-serif",
                      "opacity": 1,
                    }
                  }
                  className="withRouter-Navigation--link-6"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/admin/location"
                >
                  <a
                    className="withRouter-Navigation--link-6"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Locations
                  </a>
                </Link>
              </li>
              <li>
                <Link
                  activeStyle={
                    Object {
                      "color": "#00ba81",
                      "fontFamily": "\\"proxima-nova-bold\\", Georgia, sans-serif",
                      "opacity": 1,
                    }
                  }
                  className="withRouter-Navigation--link-6"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/admin/blog"
                >
                  <a
                    className="withRouter-Navigation--link-6"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Blog Posts
                  </a>
                </Link>
              </li>
              <li>
                <Link
                  activeStyle={
                    Object {
                      "color": "#00ba81",
                      "fontFamily": "\\"proxima-nova-bold\\", Georgia, sans-serif",
                      "opacity": 1,
                    }
                  }
                  className="withRouter-Navigation--link-6"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/admin/events"
                >
                  <a
                    className="withRouter-Navigation--link-6"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Events
                  </a>
                </Link>
              </li>
            </ul>
            <h3
              className="withRouter-Navigation--title-2 m-top-16"
            >
              Status
            </h3>
            <p
              className="withRouter-Navigation--description-3"
            >
              Not Published
            </p>
            <button
              className="btn btn__green"
              onClick={[Function]}
            >
              Publish
            </button>
          </div>
        </Navigation>
      </withRouter(Navigation)>
    </WithStyles(withRouter(Navigation))>
  </Connect(WithStyles(withRouter(Navigation)))>
</Provider>
`;
